### PR TITLE
ebpf: fix old pid_ns resolution when switch namespaces

### PR DESCRIPTION
**Pull request by xjas**
Signed-off-by: xjas <pdxzdc@foxmail.com>

## Initial Checklist

- [x] There is an issue describing the need for this PR.
- [x] Git log contains summary of the change.
- [x] Git log contains motivation and context of the change.
- [ ] If part of an EPIC, PR git log contains EPIC number.
- [ ] If part of an EPIC, PR was added to EPIC description.

## Description (git log)

when switch namespaces, use task->nsproxy to get the right old pid_ns_id

Fixes: #1972

## Type of change

- [x] Bug fix (non-breaking change fixing an issue, preferable).
- [ ] Quick fix (minor non-breaking change requiring no issue, use with care)
- [x] Code refactor (code improvement and/or code removal)
- [ ] New feature (non-breaking change adding functionality).
- [ ] Breaking change (cause existing functionality not to work as expected).

## How Has This Been Tested?

Tests being included in this PR:

- [ ] Test File A
- [ ] Test File B

Reproduce the test by running:

- command 01
- command 02

## Final Checklist:

Pick "Bug Fix" or "Feature", delete the other and mark appropriate checks.

- [ ] I have made corresponding changes to the documentation.
- [x] My code follows the style guidelines (C and Go) of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented all functions/methods created explaining what they do.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generate no new warnings.
- [ ] I have added tests that prove my fix, or feature, is effective.
- [x] New and existing unit tests pass locally with my changes.
- [x] Any dependent changes have been merged and published before.

## Git Log Checklist:

My commits logs have:

- [x] Subject starts with "subsystem|file: description".
- [x] Do not end the subject line with a period.
- [x] Limit the subject line to 50 characters.
- [x] Separate subject from body with a blank line.
- [x] Use the imperative mood in the subject line.
- [x] Wrap the body at 72 characters.
- [x] Use the body to explain what and why instead of how.

### CODE REVIEW

File: pkg/ebpf/c/tracee.bpf.c [hunk 1]
```diff
@@ -795,6 +795,11 @@ static __always_inline u32 get_task_mnt_ns_id(struct task_struct *task)
     return get_mnt_ns_id(READ_KERN(task->nsproxy));
 }
 
+static __always_inline u32 get_task_nsproxy_pid_ns_id(struct task_struct *task)
+{
+    return get_pid_ns_id(READ_KERN(task->nsproxy));
```
Comment (yanivagman) [inline pkg/ebpf/c/tracee.bpf.c:800]: This function should probably be renamed as well to `get_pid_ns_for_children_id`

Comment (yanivagman) [reply pkg/ebpf/c/tracee.bpf.c:800]: So what I actually meant was to rename the two functions:
`get_task_pid_ns_for_children_id` (the outer function youv'e added) and `get_pid_ns_for_children_id`  (this function call)

Comment (xjas) [reply pkg/ebpf/c/tracee.bpf.c:800]: Sorry to misunderstand, will fix it soon.

Comment (xjas) [reply pkg/ebpf/c/tracee.bpf.c:800]: > So what I actually meant was to rename the two functions:
> 
> `get_task_pid_ns_for_children_id` (the outer function youv'e added) and `get_pid_ns_for_children_id`  (this function call)

Sorry to misunderstand, will fix it soon.

Comment (xjas) [reply pkg/ebpf/c/tracee.bpf.c]: > So what I actually meant was to rename the two functions: `get_task_pid_ns_for_children_id` (the outer function youv'e added) and `get_pid_ns_for_children_id` (this function call)

it has been fixed, please help to have a look when you are convinent, tks.

**Review by yanivagman — APPROVED** [review]
LGTM, thanks @xjas

**Pull request merged**